Modules/Core/ImageFunction/include/itkMedianImageFunction.hxx

```diff
@@ -34,6 +35,19 @@ MedianImageFunction<TInputImage, TCoordRep>::MedianImageFunction()
 {}
 
 
+template <typename TInputImage, typename TCoordRep>
+void
+MedianImageFunction<TInputImage, TCoordRep>::SetNeighborhoodRadius(const unsigned int radius)
+{
+  if (m_NeighborhoodRadius != radius)
+  {
+    m_NeighborhoodOffsets = Experimental::GenerateRectangularImageNeighborhoodOffsets(ImageSizeType::Filled(radius));
```

Comment:
I see now, there should have been a `GenerateSquareImageNeighborhoodOffsets(unsigned radius)` helper function, creating a _square_ neighborhood shape, instead of any rectangular shape. Then the `ImageSizeType::Filled(radius)`call would not have been necessary anymore.

Reply:
`GenerateSquareImageNeighborhoodOffsets` would have been to specialized. There is a maintenance burden associated with more code. Rectangular + `ImageSizeType::Filled(radius)` is not hard to write.

Reply:
@dzenanz  I see that there's a trade-off here. It appears to me that the majority of neighborhood shapes are just square (or cubic, in 3-D). Please have a look at pull request #1347 and reject it, if you still believe the maintenance burden would be too much!